### ISSUE TITLE
Stop indexing exception objects

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5322]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5322|
|Type|enhancement|
|Created|2011-10-19 13:22:18Z|
|Branch|https://github.com/twisted/twisted/tree/dont-index-exceptions-5322|

Exception object are not directly indexable in 3.x, index the ``args`` attribute instead. Patch attached.

Attachments:

* [indexing_exceptions.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f92/f92378acbe5151c95d76c38716e617cdd44bec1c/9e32df0535a938be3af9fcf9f6691c4be3453353.patch) (1715 bytes) - added by antoine on 2011-10-19 13:22:24Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5322 5322
type__enhancement enhancement
reporter__antoine antoine
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_dont_index_exceptions_5322 branches/dont-index-exceptions-5322
branch_author__teratorn teratorn
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1319030538000000 1319030538000000
changetime__1320100175000000 1320100175000000
version__None None
owner__teratorn teratorn
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set owner to @teratorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_4"></a>|@teratorn commented|
|-|-|

(In [33023]) Branching to 'dont-index-exceptions-5322'

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_5"></a>|@teratorn commented|
|-|-|

(In [33024]) Don't index Exception objects directly as it's not supported on Python 3
Refs: [#5322](https://github.com/twisted/twisted/issues/5322)

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_6"></a>|@teratorn commented|
|-|-|

Looks good. +1. Will merge if/when the [build](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/dont-index-exceptions-5322) goes green.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_7"></a>|@teratorn commented|
|-|-|

(In [33025]) news file Refs: [#5322](https://github.com/twisted/twisted/issues/5322)

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_8"></a>|@teratorn set status to `closed`|
|-|-|

(In [33026]) Merge dont-index-exceptions-5322: Don't directly index Exception objects as it's not supported on Python 3
Author: Antoine Pitrou
Reviewer: teratorn
Fixes: [#5322](https://github.com/twisted/twisted/issues/5322)
